pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+                            else:
+                                # if it is not fixed, record its perturbed value
+                                if self.mode =='sequential_sipopt':
+                                    perturb_value = getattr(m_sipopt.sens_sol_state_1)[getattr(m_sipopt, measure_name)[0, measure_index_doublequotes,t]]
+                                else:
+                                    perturb_value = getattr(m_sipopt, measure_name)[0, measure_index_doublequotes, t]
+                            # base case values
+                            if self.mode == 'sequential_sipopt':
+                                base_value = getattr(m_sipopt, measure_name)[0, measure_index_doublequotes, t]
+                            else:
+                                base_value = getattr(mod, measure_name)[0, measure_index_doublequotes, t]
+                            perturb_mea.append(perturb_value)
+                            base_mea.append(base_value)
+
+                    else:
+                        # fetch the measurement variable
+                        measure_var = getattr(m_sipopt, j)
+                        for t in self.flatten_measure_timeset[j]:
+                            if (measure_var[0,t].fixed == True):
+                                perturb_value = value(measure_var[0, t])
+                            else:
+                                # if it is not fixed, record its perturbed value
+                                if self.mode == 'sequential_sipopt':
+                                    perturb_value = getattr(m_sipopt.sens_sol_state_1)[getattr(m_sipopt, j)[0,t]]
+                                else:
+                                    perturb_value = getattr(m_sipopt, j)[0,t]
+
+                            # base case values
+                            if self.mode == 'sequential_sipopt':
+                                base_value = pyo.value(getattr(m_sipopt, j)[0,t])
+                            else:
+                                base_value = pyo.value(getattr(mod,j)[0,t])
+
+                            perturb_mea.append(perturb_value)
+                            base_mea.append(base_value)
+
+                # store extracted measurements
+                all_perturb_measure.append(perturb_mea)
+                all_base_measure.append(base_mea)
+
+            # After collecting outputs from all scenarios, calculate sensitivity
+            for count, para in enumerate(self.param_name):
+                list_jac = []
+                for i in range(len(all_perturb_measure[0])):
+                    if self.scale_nominal_param_value:
+                        sensi = -(all_perturb_measure[count][i] - all_base_measure[count][i]) / self.step * self.scale_constant_value
+                    else:
+                        sensi = -(all_perturb_measure[count][i] - all_base_measure[count][i]) / self.step /self.param_init[para] * self.scale_constant_value
```

Comment:
This would be more clear if written as:
```python
sensi = -(all_perturb_measure[count][i] - all_base_measure[count][i]) / self.step * self.scale_constant_value
if not self.scale_nominal_param_value:
    sensi /= self.param_init[para]
```

There are other spots in the code where you could make use of the same idea.